### PR TITLE
[DW-432] Implement backend of Job Role doctype

### PR DESCRIPTION
**Pull request by DamZiobro**
+ [DW-1055] Add JobRole to person and blog doctype

### CODE REVIEW

File: site/src/main/java/uk/nhs/digital/website/beans/Role.java [hunk 1]
```diff
@@ -17,21 +15,34 @@ public String getSectionType() {
...
+    }
+
     @HippoEssentialsGenerated(internalName = "website:primaryrole")
     public String getPrimaryrole() {
+        JobRole primaryRolePicker = (JobRole)this.getPrimaryrolepicker();
+        if (primaryRolePicker != null) {
+            return primaryRolePicker.getTitle();
+        }
         return getProperty("website:primaryrole");
     }
```
Comment (GeoffHayward) [inline site/src/main/java/uk/nhs/digital/website/beans/Role.java]: Does this method deviate from the auto generated bean? If it does I normally add `allowModifications = false` to the `@HippoEssentialsGenerated` annotation. I think doing so should stop your modification being lost.

File: site/src/main/java/uk/nhs/digital/website/beans/Role.java [hunk 2]
```diff
@@ -17,21 +15,34 @@ public String getSectionType() {
...
 
+    @HippoEssentialsGenerated(internalName = "website:primaryroleorgpicker")
+    public HippoBean getPrimaryroleorgpicker() {
+        return getLinkedBean("website:primaryroleorgpicker", HippoBean.class);
+    }
+
     @HippoEssentialsGenerated(internalName = "website:primaryroleorg")
```
Comment (GeoffHayward) [inline site/src/main/java/uk/nhs/digital/website/beans/Role.java]: Does this method deviate from the auto generated bean? If it does I normally add allowModifications = false to the @HippoEssentialsGenerated annotation. I think doing so should stop your modification being lost.

File: site/src/main/java/uk/nhs/digital/website/beans/Role.java (hunk 1 above)
Comment (DamZiobro) [reply site/src/main/java/uk/nhs/digital/website/beans/Role.java]: Yes - it looks like adding allowModifications=false seems to be reasonable here as we don't want BeanWriter to modify this function any way.

Comment (DamZiobro) [reply site/src/main/java/uk/nhs/digital/website/beans/Role.java]: I added allowModifications=false, tested and pushed updated changes.

**Pull request merged**